### PR TITLE
Adding enable_sso operation in compute operation

### PR DESCRIPTION
**Pull request by achauhan-scc**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_compute_operations.py [hunk 1]
```diff
@@ -416,6 +421,26 @@ def list_sizes(self, *, location: Optional[str] = None, compute_type: Optional[s
             ]
         return [VmSize._from_rest_object(item) for item in size_list.value]
 
+    @distributed_trace
+    @monitor_with_activity(ops_logger, "Compute.enablesso", ActivityType.PUBLICAPI)
+    @experimental
+    def enable_sso(self, *, name: str, enable_sso: Optional[bool] = True):
```
Comment (diondrapeck) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_compute_operations.py:427]: "Optional" in Python means that the value can be None, but I think in this case you mean that a value doesn't necessarily have to be provided by the user. Instead, you can just do `enable_sso: bool = True`.

**Review by diondrapeck — CHANGES_REQUESTED** [review]

**Review by kingernupur — APPROVED** [review]

**Review by Garfield-ZZZZZ — APPROVED** [review]

**Review by diondrapeck — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_compute_operations.py [hunk 2]
```diff
@@ -39,11 +42,13 @@ def __init__(
         operation_scope: OperationScope,
         operation_config: OperationConfig,
         service_client: ServiceClient022023Preview,
+        service_client_2024: ServiceClient042024Preview,
         **kwargs: Dict,
     ) -> None:
         super(ComputeOperations, self).__init__(operation_scope, operation_config)
         ops_logger.update_info(kwargs)
         self._operation = service_client.compute
+        self._operation2024 = service_client_2024.compute
```
Comment (needuv) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_compute_operations.py:51]: why not change all of compute to be on the new API version? Anything that is GA in a previous version will be GA'd in the new preview version, so this should be a safe upgrade.

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_compute_operations.py [hunk 3]
```diff
@@ -416,6 +421,26 @@ def list_sizes(self, *, location: Optional[str] = None, compute_type: Optional[s
             ]
         return [VmSize._from_rest_object(item) for item in size_list.value]
 
+    @distributed_trace
+    @monitor_with_activity(ops_logger, "Compute.enablesso", ActivityType.PUBLICAPI)
+    @experimental
+    def enable_sso(self, *, name: str, enable_sso: bool = True):
```
Comment (needuv) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_compute_operations.py:427]: is there a spec for this change? This API feels awkward to me:

`client.compute.enable_sso(enable_sso=True/False)`

The verb `enable_sso` suggest something will always be turned on, but we can actually turn it off using the same `enable_sso` verb.

**Review by needuv — DISMISSED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_compute_operations.py (hunk 3 above)
Comment (achauhan-scc) [reply sdk/ml/azure-ai-ml/azure/ai/ml/operations/_compute_operations.py:427]: its specifically for data scientist role who don't have write access to compute instance, rest of role can use write/update API to perform same operation.

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_compute_operations.py (hunk 2 above)
Comment (achauhan-scc) [reply sdk/ml/azure-ai-ml/azure/ai/ml/operations/_compute_operations.py:51]: I do see this pattern of different version clients used, I am not comfortable making the changes to other API close to release cycle, I am adding a work item and resolving your comment as this point of time.
[Task 3372541](https://msdata.visualstudio.com/Vienna/_workitems/edit/3372541): Consolidate the rest client version for Compute operations.

**Pull request merged**